cpp/include/cudf/io/orc.hpp

```diff
@@ -42,6 +42,30 @@ constexpr size_t default_stripe_size_bytes   = 64 * 1024 * 1024;  ///< 64MB defa
 constexpr size_type default_stripe_size_rows = 1000000;  ///< 1M rows default orc stripe rows
 constexpr size_type default_row_index_stride = 10000;    ///< 10K rows default orc row index stride
 
+/**
+ * @brief Check if the compression type is supported for reading ORC files.
+ *
+ * @note This is a runtime check. Some compression types may not be supported because of the current
+ * system configuration.
+ *
+ * @param compression Compression type
+ * @return true if supported
+ * @return false if not supported
+ */
+[[nodiscard]] bool is_compressed_read_orc_supported(compression_type compression);
```

Comment:
I think the name is not accurate, since we are checking a specific the compression type, not all compression.
```suggestion
[[nodiscard]] bool is_compression_type_supported_read_orc(compression_type compression);
```

Reply:
I almost used this one. Would not mind changing to it.